services/streammanager/googlecloudfunction/googlecloudfunction.go

```diff
@@ -0,0 +1,261 @@
+
+	return &GoogleCloudFunctionProducer{client, destConfig}, err
+}
+
+func (producer *GoogleCloudFunctionProducer) Produce(jsonData json.RawMessage, _ interface{}) (statusCode int, respStatus, responseMessage string) {
+	destConfig := producer.config
+	parsedJSON := gjson.ParseBytes(jsonData)
+
+	if destConfig.FunctionEnvironment == "gen1" {
+		return invokeGen1Functions(producer.client, destConfig.FunctionName, parsedJSON)
+	}
+
+	return invokeGen2Functions(destConfig.GoogleCloudFunctionUrl, destConfig.Credentials, destConfig.RequireAuthentication, parsedJSON)
+}
+
+func invokeGen1Functions(client *Client, functionName string, parsedJSON gjson.Result) (statusCode int, respStatus, responseMessage string) {
+	if client == nil {
+		respStatus = "Failure"
+		responseMessage = "[GoogleCloudFunction] error  :: Failed to initialize GoogleCloudFunction client"
+		return 400, respStatus, responseMessage
+	}
+
+	requestPayload := &cloudfunctions.CallFunctionRequest{
+		Data: parsedJSON.String(),
+	}
+
+	// Make the HTTP request
+	call := client.service.Projects.Locations.Functions.Call(functionName, requestPayload)
+
+	response, err := call.Do()
+
+	if err != nil {
+		statCode, serviceMessage := handleServiceError(err)
+		respStatus = "Failure"
+		responseMessage = "[GOOGLE_CLOUD_FUNCTION] error :: Function call was not executed (Not Modified :: " + serviceMessage
+		pkgLogger.Errorf("[GOOGLE_CLOUD_FUNCTION] error while calling the Gen1 function :: %v", err)
+		return statCode, respStatus, responseMessage
+	}
+
+	// ---------- To be Removed
+	// Process the response (sample response handling here).
+	if response != nil {
+		fmt.Printf("Function call status code: %d\n", response.HTTPStatusCode)
+		// Handle response content as needed.
+	}
+	fmt.Println("Request successful!")
+	// -----------
+
+	respStatus = "Success"
+	responseMessage = "[GoogleCloudFunction] :: Message Payload inserted with messageId :: " + parsedJSON.Get("id").String()
+	return 200, respStatus, responseMessage
+}
+
+func invokeGen2Functions(functionUrl, credentials string, requireAuthentication bool, parsedJSON gjson.Result) (statusCode int, respStatus, responseMessage string) {
+	ctx := context.Background()
+
+	jsonBytes := []byte(parsedJSON.String())
+
+	// Create a POST request
+	req, err := http.NewRequest("POST", functionUrl, strings.NewReader(string(jsonBytes)))
+	if err != nil {
+		fmt.Println("Error creating request:", err)
+		return
+	}
+
+	// Set the appropriate headers
+	req.Header.Set("Content-Type", "application/json")
+	if requireAuthentication {
+		ts, err := idtoken.NewTokenSource(ctx, functionUrl, option.WithCredentialsJSON([]byte(credentials)))
+		if err != nil {
+			fmt.Print("failed to create NewTokenSource: %w", err)
+		}
+
+		// Get the ID token, to make an authenticated call to the target audience.
+		token, err := ts.Token()
+		if err != nil {
+			fmt.Print("failed to receive token: %w", err)
```

Comment:
I think we should handle this error in a different way. Are we going to fail the event & make sure it re-tries ? or Is there any other approach planned for this ?